main.go

```diff
@@ -337,28 +339,29 @@ func initConsensus(ctx *cli.Context, p2pPid *actor.PID, txpoolSvr *proc.TXPoolSe
 	return consensusService, nil
 }
 
-func initRpc(ctx *cli.Context) error {
+func initRpc(ctx *cli.Context) (aux.Stopper, error) {
 	if !config.DefConfig.Rpc.EnableHttpJsonRpc {
-		return nil
+		return nil, nil
 	}
+	var stopper aux.Stopper
 	var err error
 	exitCh := make(chan interface{}, 0)
 	go func() {
-		err = jsonrpc.StartRPCServer()
+		stopper, err = jsonrpc.StartRPCServer()
 		close(exitCh)
 	}()
 
 	flag := false
 	select {
 	case <-exitCh:
 		if !flag {
-			return err
+			return nil, err
 		}
 	case <-time.After(time.Millisecond * 5):
```

Comment:
This is a dangerous workaround. Please fix it with sync primitives instead of sleep if the current task covers this peace of code.